code/modules/mob/living/carbon/human/death.dm

```diff
@@ -104,6 +104,10 @@
 	//Handle species-specific deaths.
 	if(species) species.handle_death(src)
 
+	var/obj/item/organ/internal/lantern/L = get_int_organ(/obj/item/organ/internal/lantern)

```

Comment:
It's not just humans that can have organs.

My idea was you can make a proc called `on_owner_death` that is called when a carbon dies.

Something like:

```
/obj/item/organ/proc/on_owner_death()
 	return
```
then have `/obj/item/organ/internal/lantern` do this:

```
/obj/item/organ/internal/lantern/on_owner_death()
	if(glowing)
		togglebiolum(1)
```

Finally, in carbon death, you'd have something like this:

```
for(var/obj/item/organ/O in organs)
 	O.on_owner_death()
```

This is more OOP, and allows for modular things like organs that cause you to explode when you die, or auto-revives and the likes.